docs/en/routing/basic-usage.md

```diff
@@ -0,0 +1,490 @@
+* `type: 'index' | 'partial' | 'error'`: The type of match for the route, either `index`, `partial` or `error`.
+* `router: RouterInterface`: The router instance which can used to create links and initiate route changes.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home',
+		outlet: 'home',
+		children: [
+			{
+				path: 'details',
+				outlet: 'home-details'
+			}
+		]
+	}
+];
+```
+
+>src/App.tsx
+```tsx
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import Outlet from '@dojo/framework/routing/Outlet';
+
+export default class App extends WidgetBase {
+	protected render() {
+		return (
+			<div>
+				<Outlet id="home" renderer={(matchDetails) => {
+					const { params, queryParams, isExact, isError, router } = matchDetails;
+
+					const gotoHome = () => {
+						const link = router.link('home');
+						if (link) {
+							router.setPath(link);
+						}
+					};
+
+					if (isExact()) {
+						// The path `home` was matched
+						return <div>Home Page</div>
+					}
+					if (isError()) {
+						// The `home` segment of the path was matched but the next segment was not match
+						// for example, `home/other`
+						return (
+							<div>
+								<button onclick={gotoHome}>Goto Home</button>
+								<div>Unknown Page</div>
+							</div>
+						);
+					}
+					// The `home` segment of the path was matched and the next segment was also matched
+					// for example, `home/details`
+					return <div>Partial Match for Home</div>
+
+				}} />
+			</div>
+		);
+	}
+}
+```
+
+## `Link` and `ActiveLink`
+
+The `Link` component is a wrapper around an `a` DOM element that enables consumers to specify an `outlet` to create a link to. It is also possible to use a static route by setting the `isOutlet` property to `false`.
+
+If the generated link requires specific path or query parameters that are not in the route, they can be passed via the `params` property.
+
+Link Properties:
+
+* `to: string`: The `outlet` id or `href` for the link.
+* `params: { [index: string]: string }`: Params to generate the link with for the outlet.
+* `isOutlet: boolean`(optional): Indicates if the link is for an outlet or a static `href`, defaults to `true`.
+* `onClick: (event: MouseEvent) => void`(optional): Function that gets called when the `Link` is clicked.
+
+In addition to the `Link` specific properties, all the standard `VNodeProperties` are available for the `Link` component as they would be creating an `a` DOM Element.
+
+>src/App.tsx
+```tsx
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import { Link } from '@dojo/framework/routing/Link';
+
+export default class App extends WidgetBase {
+	protected render() {
+		return (
+			<div>
+				<Link to="home" params={{ foo: 'bar' }}>Link Text</Link>
+				<Link to="#/static-route" isOutlet={false}>Link Text</Link>
```

Comment:
should the hash be here? doesn't this make it coupled to the history?

Reply:
You can use `Link` to set a static `href` and not generate the link via the router based on an outlet... So if you did it like this then it would need to have the `#` prefix, however I'm not sure I see the value in supporting this with `Link`.

Reply:
Yeah this functionality doesn't even make sense to me if it doesn't at least decouple the history. This gives me no advantage over an anchor tag

Reply:
Aye also my thoughts, I'll remove this from the ref guide.

Reply:
As it's confusing if nothing else.

Reply:
Removed